### ISSUE TITLE
Failed initialization prohibits startup

### ISSUE THREAD

**Issue by hannesmuehleisen** [hannesmuehleisen/MonetDBLite#17]
Steps to Reproduce:
```R
# # # R session one

library(DBI)
library(MonetDB.R)
library(MonetDBLite)

# you will need this same filepath in session two
( my_tempdir <- tempdir() )

db <- dbConnect( MonetDBLite() , my_tempdir )

# copy this folder path to clipboard
dput( my_tempdir )

# # # keep R session one open

# # # R session two

library(DBI)
library(MonetDB.R)
library(MonetDBLite)

# use the same `my_tempdir` as session one
my_tempdir <- [[use output of session one `dput` here]]

# expected error:
db <- dbConnect( MonetDBLite() , my_tempdir )
# Error in MonetDBLite::monetdb_embedded_startup(embedded, !getOption("monetdb.debug.embedded",  : 
  # Failed to initialize embedded MonetDB !FATAL: GDKlockHome: Database lock '.gdk_lock' denied

# # # CLOSE R session one

# # # remain in R session two

# still does not connect to database created in session one
db <- dbConnect( MonetDBLite() , my_tempdir )
# Error in MonetDBLite::monetdb_embedded_startup(embedded, !getOption("monetdb.debug.embedded",  : 
  # Failed to initialize embedded MonetDB !FATAL: BBPaddfarm: bad rolemask

  
# also does not connect to a different database..  
db <- dbConnect( MonetDBLite() , tempdir() )
# Error in MonetDBLite::monetdb_embedded_startup(embedded, !getOption("monetdb.debug.embedded",  : 
  # Failed to initialize embedded MonetDB !FATAL: BBPaddfarm: bad rolemask

```

**Comment by monicarodguev** [comment]
@hannesmuehleisen did you solve the error?

**Comment by hannesmuehleisen** [comment]
no not yet unfortunately.

**Comment by monicarodguev** [comment]
What we are thinking is happening is the bbdd got corrupted after forcing a query to stop. But is just our intuition.

**Comment by teshomem** [comment]
@hannesmuehleisen i think this a major issue using monetdb in shiny app.

**Comment by solmanz** [comment]
I am not sure if this is related. MonetDB doesn't work when I start shiny application from shiny server (browser). I get "ERROR: Failed to initialize embedded MonetDB !FATAL: BBPaddfarm: bad rolemask". It works when i start the shiny app from RStudio.

**Comment by hannesmuehleisen** [comment]
I have no experience with Shiny Server, but assume its trying to connect multiple R sessions to the same MonetDBLite folder?

**Comment by solmanz** [comment]
Yes, it is trying to connect multiple R sessions to same MonetDBLite directory. I have around 15GB csv formatted genomic data for visualisation (common issue in genomics field). I have tested different database solution including data table. MonetDB is the right and fastest database for this kind of shiny applications if it works. If this is fixable, I can write a white paper to show how MonetDBLite should be a choice for growing genomics data visualisation in R. I don't know what is it going to take to fix this but I can assure you that making MonetDBLite friendly with shiny will make it very popular in short period of time.

**Comment by ajdamico** [comment]
hi @solmanz just in case it's of any use: you can connect multiple R sessions to a single external mserver with `?MonetDBLite::MonetDB.R`

**Comment by solmanz** [comment]
Thanks @ajdamico but i used that for connection when i get the error message. The problem is I don't know the root cause. It could be multiple R session or another.

**Comment by ajdamico** [comment]
i'm not sure this fix is high priority if you prefer to migrate to 0.5.0.  thanx bud

**Comment by hannesmuehleisen** [comment]
its not and difficult to fix.

**Comment by cboettig** [comment]
I get this same error if an R session crashes.  Starting a new session I can disconnect with `MonetDBLite::monetdblite_shutdown()` to free the `.gdk_lock`, and then I get this same error.  I gather this means the database is corrupt at this point?  Is recovery of the data possible?

**Comment by hannes** [comment]
@cboettig which error exactly?

**Comment by cboettig** [comment]
@hannesmuehleisen sorry I wasn't precise, thanks for the followup.  After removing the `.gdk_lock` after the crash, I get the error:

`ERROR: Failed to initialize embedded MonetDB !FATAL: BBPaddfarm: bad rolemask"`

whenever I try to connect to my database.

**Comment by hannes** [comment]
@cboettig thats not great. Could you please open up a separate issue, since it seems unrelated to the issue discussed here.

**Comment by cboettig** [comment]
@hannesmuehleisen ok, thanks, will do.  Would you prefer the issue here or over in https://github.com/hannesmuehleisen/MonetDBLite-R ?

**Comment by hannes** [comment]
@cboettig here is fine, also include some details (`sessionInfo()` output) and if you can a reproducible example please.
